src/ufm/fine_tuning.py

```diff
@@ -40,30 +43,42 @@ def validate_finetune_cfg(cfg: DictConfig) -> DictConfig:
     return cfg
 
 
-def run_fine_tune(
-    # model_unadapted: HuggingFaceModel,
-    model_unadapted,
+def run_llm_fine_tune(
+    model: Union["UFMLangaugeModel", "nn.Module"],
     config: dict,
-    # training_task: str = "supervised-fine-tuning",
-):
+    tokenizer: Optional[Union[PreTrainedTokenizerBase, AutoTokenizer]] = None,
+) -> Dict[str, list]:
     """
-    Fine tunes model_unadapted on the dataset specified in config
+    Finetunes a model on the dataset specified in config
     Returns fine-tuning validation losses
     Only supports supervised-fine-tuning for now
     Only supports text-only datasets (cyber and pile) for now
 
-    'config' is specifically the 'finetune' struct of global config
+    :param model: UFMLanguageModel or nn.Module. If nn.Module, tokenizer must be provided
+    :param config: DictConfig with finetuning config
+    :param tokenizer: Optional tokenizer. Required if model is nn.Module
+    :returns: dict of evals
     """
+    if isinstance(model, nn.Module):
+        if tokenizer is None:
+            raise ValueError(
+                f"If not UFMLanguageModel, tokenizer must be provided. Received {type(model)}."
+            )
+        llm = model
+    elif isinstance(model, UFMLangaugeModel):
+        llm = model.model
+        tokenizer = model.tokenizer
+    else:
+        # Ambiguous
+        raise ValueError(f"Model must be either UFMLanguageModel or nn.Module. Received {type(model)}.")
```

Comment:
I think we should still call it model, not llm. You can make the first 2 `if`s into a single assertion or and if and assert.